### ISSUE TITLE
Register a callee with limit for maximum "in-fly" invocations

### ISSUE THREAD

**Issue by oberstet** [crossbario/crossbar#621]
A callee might implement a registered procedure by processing incoming invocations sequentially - that is, not more than one invocation concurrently. A callee might also implement a registered procedure by processing incoming invocations concurrently - that is, more than one invocation concurrently.

For example, the callee might spawn a new background worker process or thread for each invocation to do some work, or might manage a reusable pool of background worker processes or threads to run work for incoming invocations.

In this situation, the work for invocations is effectively distributed over CPU cores. However, there is only a fixed number of CPU cores available. So when more invocations are scheduled concurrently than there are CPU cores available, the number of working processes or threads will exceed the core count, and the OS scheduling will kick in and do "fair scheduling" among all those running tasks.

The end result is, that all invocations take relatively long, but roughly the same time. If the OS scheduling is doing a good job and is able to share the CPU time available from all cores fairly among all active tasks, the tasks will finish all nearly at the same time.

Also, all the CPU cores will be used up or at least tried to use. So this might interfere with other work the host the callee is running on is supposed to do.

Further, when there is subsequent work being started directly or indirectly by the caller of above, then this can lead to the next wave of calls all issued nearly at the same time. Possibly continuing call layer per layer.

So in summary, this is bad;)

What's needed is a user configurable limit to the number of outstanding invocations for a registered callee.

**Comment by oberstet** [comment]
We have that feature now

**Comment by markope** [comment]
What happens if more invocations come in than maximum. Are they queued or rejected? Is the queue length configurable?

**Comment by oberstet** [comment]
Yes, we have queuing too: https://github.com/crossbario/crossbar-examples/tree/master/scaling-microservices
